nflows/transforms/splines/rational_quadratic.py

```diff
@@ -29,34 +29,36 @@ def unconstrained_rational_quadratic_spline(
     logabsdet = torch.zeros_like(inputs)
 
     if tails == "linear":
-        unnormalized_derivatives = F.pad(unnormalized_derivatives, pad=(1, 1))
         constant = np.log(np.exp(1 - min_derivative) - 1)
-        unnormalized_derivatives[..., 0] = constant
-        unnormalized_derivatives[..., -1] = constant
+        unnormalized_derivatives = F.pad(unnormalized_derivatives, pad=(1, 1),
+                                         value=constant)
 
-        outputs[outside_interval_mask] = inputs[outside_interval_mask]
-        logabsdet[outside_interval_mask] = 0
+        outputs += outside_interval_mask*inputs
+        # logabsdet += outside_interval_mask*0
     else:
         raise RuntimeError("{} tails are not implemented.".format(tails))
 
-    if torch.any(inside_interval_mask):
```

Comment:
Also, removed this check, as I suppose most of the time it will evaluate to true as you would expect some inputs to be in the domain. Let me know if you'd prefer it added back.

Reply:
I see this was added in https://github.com/bayesiains/nflows/pull/25. With the new implementation there won't be any crashes with all-tail inputs either. (Except the computations will essentially be wasted, but I don't suppose we're expecting many calls with all-tails inputs.)